src/main/java/seedu/duke/userinterface/command/Add.java

```diff
@@ -1,8 +1,48 @@
 package seedu.duke.userinterface.command;
 
+import seedu.duke.exceptions.IncorrectDeadlineFormatException;
+import seedu.duke.exceptions.TaskTitleException;
+import seedu.duke.exceptions.TaskWrongFormatException;
+import seedu.duke.tasks.Task;
+import seedu.duke.tasks.TaskList;
+import seedu.duke.userinterface.CliMessages;
+import seedu.duke.userinterface.InputParser;
+
 public class Add extends CliCommand {
+    public static final String COMMAND_WORD = "add";
+    public static final String TASK_DELIMITER = "/t";
+    public static final String DEADLINE_DELIMITER = "/by";
+    private final TaskList taskArrayList;
+    private String input;
+    private static InputParser parser;
+
+    public Add(TaskList taskArrayList, String input) {
+        this.taskArrayList = taskArrayList;
+        this.input = input;
+    }
+
     @Override
     public void execute() {
-
+        parser = new InputParser();
+        try {
+            if (input.contains("/by")) {
+                String title = parser.parseTaskTitle(input);
+                String deadline = parser.parseDeadline(input);
+                taskArrayList.addTask(new Task(title, deadline));
+                CliMessages.printAddedTaskMessage(taskArrayList, title);
+            } else {
+                throw new TaskWrongFormatException();
+            }
+        } catch (TaskTitleException t) {
+            System.out.println("\tYour task is missing a title!");
```

Comment:
We should add custom exceptions for all of these, catch those in CliUserInterface, and use an inherited `printErrorMessage()` method or something right?

Reply:
I'll add this as a `//TODO` for now